src/include/neorados/RADOS.hpp

```diff
@@ -127,9 +127,8 @@ class IOContext final {
 public:
 
   IOContext();
-  explicit IOContext(std::int64_t pool);
-  IOContext(std::int64_t _pool, std::string_view _ns);
-  IOContext(std::int64_t _pool, std::string&& _ns);
+  IOContext(std::int64_t pool);
```

Comment:
why remove the explicit call when ctor accept one int parameter?

Reply:
but "pool" is just ant int. this will create implicit conversion from int to IOContext.
isn't thank risky?

Reply:
I don't think so, unless I'm missing something. We don't really have any overloads it's likely to catch by accident, and conceptually IOContext (really I should have named it IOLocator so people wouldn't be tempted to stuff operation flags into it or something. Also that's what it maps to in Objector) is a pool (indicated by an int) plus some other less used stuff.

For something that's "Int plus optional other stuff" being able to implicitly accept an int doesn't seem wrong.

Reply:
i've seen such a bug in our code when we had a non-explicit ctor of `rgw_user` that accepts an int.
my concern is that we have:
```
foo(IOContext ctx, ...)
```
and someone incorrectly invokes it with an int (which is not the pool id) as the first parameter.
the compiler won't complain and implicitly construct an `IOContext` based on an integer that is not a pool id (this is very likely, if `foo` accepts multiple integer parameters, which is quite common).
making the ctor explicit, prevents these kinds of bugs.

Reply:
All right, played around with this and the ergonomics of changing it aren't too bad in practice.

Reply:
All right. Made this change and tested it out.